pkg/authorization/authorizer/scope/converter.go

```diff
@@ -244,6 +300,37 @@ func (e clusterRoleEvaluator) ResolveRules(scope, namespace string, clusterPolic
 	return rules, nil
 }
 
+func (e clusterRoleEvaluator) ResolveGettableNamespaces(scope string, clusterPolicyGetter client.ClusterPolicyLister) ([]string, error) {
+	_, scopeNamespace, _, err := e.parseScope(scope)
+	if err != nil {
+		return nil, err
+	}
+	rules, err := e.resolveRules(scope, clusterPolicyGetter)
+	if err != nil {
+		return nil, err
+	}
+
+	attributes := authorizer.DefaultAuthorizationAttributes{
+		APIGroup: kapi.GroupName,
+		Verb:     "get",
+		Resource: "namespaces",
+	}
+
+	errors := []error{}
+	for _, rule := range rules {
+		matches, err := attributes.RuleMatches(rule)
```

Comment:
> rule.ResourceNames contains scopeNamespace

It won't because the rule is only applied in a given namespace.  Admin has "get" on "namespace" and the rule is only checked if the role is bound in that namespace.


> rule.ResourceNames is empty

I think the "I said he could look at an invalid namespace name" is a small enough edge to accept.  That role never worked right.



> rule.ResourceNames contains "*"

`*` isn't special for resourceNames.  Empty list means all, `*` tries to match the name `*`.  That's my "do awesome safe things" username.